### PR TITLE
parse relayed txs

### PR DESCRIPTION
**Pull request by miiu96**
Parse `relayed` transactions

### CODE REVIEW

File: process/transactions/datafield/parseRelayed.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"encoding/json"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+)
+
+const (
+	numArgsRelayedV2 = 4
+)
+
+func (odp *operationDataFieldParser) ParseRelayed(function string, args [][]byte, receiver []byte) *ResponseParseData {
```
Comment (iulianpascalau) [inline process/transactions/datafield/parseRelayed.go:13]: missing comments on exported item(s)

File: process/transactions/datafield/parseRelayed_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			"@" +
+			"01a2")
+		res := parser.Parse(dataField, sender, receiver)
+		require.Equal(t, &ResponseParseData{
+			IsRelayed: true,
+		}, res)
+	})
+
+	t.Run("RelayedTxV2WithNFTTransfer", func(t *testing.T) {
+		nftTransferData := []byte("ESDTNFTTransfer@4c4b4641524d2d396431656138@34ae14@728faa2c8883760aaf53bb@000000000000000005001e2a1428dd1e3a5146b3960d9e0f4a50369904ee5483@636c61696d5265776172647350726f7879@00000000000000000500a655b2b534218d6d8cfa1f219960be2f462e92565483")
```
Comment (dragos-rebegea) [inline process/transactions/datafield/parseRelayed_test.go:102]:
```suggestion
                t.Parallel()
                
		nftTransferData := []byte("ESDTNFTTransfer@4c4b4641524d2d396431656138@34ae14@728faa2c8883760aaf53bb@000000000000000005001e2a1428dd1e3a5146b3960d9e0f4a50369904ee5483@636c61696d5265776172647350726f7879@00000000000000000500a655b2b534218d6d8cfa1f219960be2f462e92565483")
```

Comment (miiu96) [reply process/transactions/datafield/parseRelayed_test.go:102]: done.

File: process/transactions/datafield/parseRelayed.go (hunk 1 above)
Comment (miiu96) [reply process/transactions/datafield/parseRelayed.go]: refactored.

**Review by iulianpascalau — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**